library/aes.c

```diff
@@ -1824,6 +1824,13 @@ int mbedtls_aes_self_test(int verbose)
         } else
 #endif
 #if defined(MBEDTLS_AESNI_HAVE_CODE)
+#if MBEDTLS_AESNI_HAVE_CODE == 1
+            mbedtls_printf("  AES note: AESNI code present (assembly implementation).\n");
+#elif MBEDTLS_AESNI_HAVE_CODE == 2
+            mbedtls_printf("  AES note: AESNI code present (intrinsics implementation).\n");
+#else
+#error Unrecognised value for MBEDTLS_AESNI_HAVE_CODE
+#endif
```

Comment:
Just a question about this block.

It seems that it would change the original logic when `MBEDTLS_PADLOCK_C` and `MBEDTLS_AESNI_HAVE_CODE` were defined. Was this behavior expected?

Reply:
How does this change the logic?

Reply:
Previously, when `MBEDTLS_PADLOCK_C`, `MBEDTLS_AESNI_HAVE_CODE` and `MBEDTLS_HAVE_X86 ` were defined, here we had:
```C
        if (mbedtls_padlock_has_support(MBEDTLS_PADLOCK_ACE)) {
            mbedtls_printf("  AES note: using VIA Padlock.\n");
        } else
        if (mbedtls_aesni_has_support(MBEDTLS_AESNI_AES)) {
            mbedtls_printf("  AES note: using AESNI.\n");
        } else
        mbedtls_printf("  AES note: built-in implementation.\n");
```
But with this block, we would have:
```C
        if (mbedtls_padlock_has_support(MBEDTLS_PADLOCK_ACE)) {
            mbedtls_printf("  AES note: using VIA Padlock.\n");
        } else
            mbedtls_printf("  AES note: AESNI code present (intrinsics implementation).\n");

        if (mbedtls_aesni_has_support(MBEDTLS_AESNI_AES)) {
            mbedtls_printf("  AES note: using AESNI.\n");
        } else
        mbedtls_printf("  AES note: built-in implementation.\n");
```

It only affects the build when padlock and aesni are enabled for x86 32bit.

Reply:
> there's no hardware that supports both VIA Padlock and AESNI

Thanks @gilles-peskine-arm, this is exactly what I am not sure. 

But it might still matter in other case. If someone ran the selftest on VIA Padlock supported machine, they would get confusing messages.